cyral/internal/policyv2/model.go

```diff
@@ -0,0 +1,158 @@
+	}
+	if err := d.Set("description", r.Policy.Description); err != nil {
+		return fmt.Errorf("error setting 'description' field: %w", err)
+	}
+	if err := d.Set("enabled", r.Policy.Enabled); err != nil {
+		return fmt.Errorf("error setting 'enabled' field: %w", err)
+	}
+	if err := d.Set("tags", r.Policy.Tags); err != nil {
+		return fmt.Errorf("error setting 'tags' field: %w", err)
+	}
+	if err := d.Set("valid_from", r.Policy.ValidFrom); err != nil {
+		return fmt.Errorf("error setting 'valid_from' field: %w", err)
+	}
+	if err := d.Set("valid_until", r.Policy.ValidUntil); err != nil {
+		return fmt.Errorf("error setting 'valid_until' field: %w", err)
+	}
+
+	if err := d.Set("document", r.Policy.Document); err != nil {
+		return fmt.Errorf("error setting 'document' field: %w", err)
+	}
+
+	if err := d.Set("last_updated", map[string]interface{}{
+		"actor":      r.Policy.LastUpdated.Actor,
+		"actor_type": r.Policy.LastUpdated.ActorType,
+		"timestamp":  r.Policy.LastUpdated.Timestamp,
+	}); err != nil {
+		return fmt.Errorf("error setting 'last_updated' field: %w", err)
+	}
+	if err := d.Set("created", map[string]interface{}{
+		"actor":      r.Policy.Created.Actor,
+		"actor_type": r.Policy.Created.ActorType,
+		"timestamp":  r.Policy.Created.Timestamp,
+	}); err != nil {
+		return fmt.Errorf("error setting 'created' field: %w", err)
+	}
+	if err := d.Set("enforced", r.Policy.Enforced); err != nil {
+		return fmt.Errorf("error setting 'enforced' field: %w", err)
+	}
+	if r.Policy.Type != "" {
+		if err := d.Set("type", r.Policy.Type); err != nil {
+			return fmt.Errorf("error setting 'type' field: %w", err)
+		}
+	}
+	if err := d.Set("scope", flattenScope(r.Policy.Scope)); err != nil {
+		return fmt.Errorf("error setting 'scope' field: %w", err)
+	}
+	d.SetId(r.Policy.ID)
+	return nil
+}
+
+// ReadFromSchema reads the policy data from the schema
+func (r *PolicyV2) ReadFromSchema(d *schema.ResourceData) error {
+	r.Policy.ID = d.Get("id").(string)
+	r.Policy.Name = d.Get("name").(string)
+	r.Policy.Description = d.Get("description").(string)
+	r.Policy.Enabled = d.Get("enabled").(bool)
+	r.Policy.Tags = expandStringList(d.Get("tags").([]interface{}))
+	r.Policy.ValidFrom = d.Get("valid_from").(string)
+	r.Policy.ValidUntil = d.Get("valid_until").(string)
+	r.Policy.Document = d.Get("document").(string)
+	r.Policy.Enforced = d.Get("enforced").(bool)
+	r.Policy.Type = d.Get("type").(string)
+	if v, ok := d.GetOk("scope"); ok {
+		r.Policy.Scope = expandScope(v.([]interface{}))
+	}
+	return nil
+}
+
+// expandStringList converts a list of interface{} to a list of strings
+func expandStringList(list []interface{}) []string {
+	result := make([]string, len(list))
+	for i, v := range list {
+		result[i] = v.(string)
+	}
+	return result
+}
+
+// getStringOrDefault returns the string value or an empty string if nil or not a string
+func getStringOrDefault(v interface{}) string {
+	str, ok := v.(string)
+	if !ok {
+		return ""
+	}
+	return str
```

Comment:
Slightly more idiomatic and compact would be the following (feel free to ignore):
```suggestion
	if str, ok := v.(string); ok {
		return str
	}
	return ""
```